File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int i,j,n=nums.size(),flag=0;
        i=0,j=n-1;
        map<int,int> m;
        for(i=0;i<n;i++)
            m[nums[i]]=i+1;
        for(i=0;i<n;i++)
            if(m[target-nums[i]] && m[target-nums[i]]!=m[nums[i]])
                break;

        vector<int> ans{m[nums[i]],m[target-nums[i]]};
        return ans;
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> store = new HashMap<>();
        for(int i=0;i<nums.length;i++){
            if(store.getOrDefault((target-nums[i]),-1)!=-1){
                return new int[]{store.get(target-nums[i]),i};
            }
            store.put(nums[i], i);
        }
        return new int[]{};
    }
}
